.github/workflows/release.yml

```diff
@@ -39,6 +39,8 @@ jobs:
             tag="latest"
           fi
           echo "VERSION=$tag" >> $GITHUB_OUTPUT
+      - name: Build docker image
+        run: make image CONTAINER_TOOL=docker IMAGE_FULL_PATH=${{ secrets.QUAYIO_ORG }}/numaplane-controller:${{ steps.version.outputs.VERSION }}
       - name: Container build and push with arm64/amd64
         run: make image docker-push IMAGE_FULL_PATH=${{ secrets.QUAYIO_ORG }}/numaplane-controller:${{ steps.version.outputs.VERSION }} CONTAINER_TOOL=docker
```

Comment:
we don't need to push the image for each pull request or push, it should only push the image if it's a release tag or PR merged to `main` branch.

Reply:
So, this Workflow only gets invoked when it’s a push to main branch or release tag.

Reply:
But I feel this PR is not needed at all, right? Since he is already building and pushing the image in subsequent step

Reply:
Right, this PR is not needed at all

Reply:
If this PR is fixing Issue #150 then need to add image build step(without push) in ci.yaml which will run on every PR

Reply:
oh, is the idea to verify that we can build the image without it failing?

Reply:
yes

Reply:
@chandankumar4 so remove both of these line ``` - name: Build docker image
        run: make image CONTAINER_TOOL=docker IMAGE_FULL_PATH=${{ secrets.QUAYIO_ORG }}/numaplane-controller:${{ steps.version.outputs.VERSION }}
      - name: Container build and push with arm64/amd64
        run: make image docker-push IMAGE_FULL_PATH=${{ secrets.QUAYIO_ORG }}/numaplane-controller:${{ steps.version.outputs.VERSION }} CONTAINER_TOOL=docker``` from the release.yaml and add ` - name: Build docker image
        run: make image CONTAINER_TOOL=docker IMAGE_FULL_PATH=${{ secrets.QUAYIO_ORG }}/numaplane-controller:${{ steps.version.outputs.VERSION }}` to the ci.yaml?

Reply:
@chandankumar4 i made the changes. lmk what you think

Reply:
Thanks, changes looks good, just one minor comment!!